### PR TITLE
Fix dependency workflow failures

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Three fixes to the dependency workflow:

- Fix the path to the upload-to-s3 action, which was incorrect
- Add a length check for number of new versions found, to prevent GHA issues with empty matrices.
  - length-related changes have been tested here: https://github.com/paketo-buildpacks/nginx/actions/runs/3245137061 (the workflow was previously failing)
- Add logic to only run compilation/compilation-related steps on the matrix of metadata entries that need compilation (missing checksum and URI). This protects against failures we saw in https://github.com/paketo-buildpacks/cpython/actions/runs/3223469727 in which one stack-variant of `cpython` didn't need compilation and caused workflow failures
  - the changes have been tested in https://github.com/paketo-buildpacks/cpython/actions/runs/3266553148

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Review by robdimsdale — APPROVED** [review]

**Comment by sophiewigmore** [comment]
length-related changes have been tested here: https://github.com/paketo-buildpacks/nginx/actions/runs/3245137061
this resolves failures we saw in the Nginx GHAs, due to no new versions being found the compilation step was erroring due to  retrieval metadata matrix being empty.

**Comment by sophiewigmore** [comment]
Workflow changes tested: 
https://github.com/paketo-buildpacks/nginx/actions/runs/3266794598/jobs/5371055424
https://github.com/paketo-buildpacks/cpython/actions/runs/3266797067

File: implementation/.github/workflows/update-dependencies-from-metadata.yml
Comment (robdimsdale) [inline implementation/.github/workflows/update-dependencies-from-metadata.yml:288]: Can we update the numbers to reflect the extra step?

File: implementation/.github/workflows/update-dependencies-from-metadata.yml
Comment (robdimsdale) [inline implementation/.github/workflows/update-dependencies-from-metadata.yml:166]: does this `if` need an `always() && ${{ ... }}`?

Comment (robdimsdale) [reply implementation/.github/workflows/update-dependencies-from-metadata.yml:166]: I think it's fine because we're not evaluating whether jobs are `skipped` - all dependent jobs must have passed, which is fine for this `if` statement.

I could see value in adding in the `always()` for consistency and future-proofing, but I can also see value in keeping it out as it is redundant.

**Review by robdimsdale — APPROVED** [review]
I have a small comment and a quick question but otherwise this looks good to me.

File: implementation/.github/workflows/update-dependencies-from-metadata.yml
Comment (sophiewigmore) [reply implementation/.github/workflows/update-dependencies-from-metadata.yml:288]: oops yes

File: implementation/.github/workflows/update-dependencies-from-metadata.yml
Comment (sophiewigmore) [reply implementation/.github/workflows/update-dependencies-from-metadata.yml:166]: What would the value be of an `always()`? here? I'm not sure I can see the need for it, if you could explain a bit more about that

Comment (sophiewigmore) [reply implementation/.github/workflows/update-dependencies-from-metadata.yml:166]: Since it has `retrieve` and `get-compile-and-test` as `needs` I think that covers our bases, no?

**Comment by sophiewigmore** [comment]
bump @paketo-buildpacks/tooling-maintainers I'd love to get this reviewed ASAP so we can unblock workflow failures.

**Review by ForestEckhardt — APPROVED** [review]

**Review by fg-j — APPROVED** [review]

**Pull request merged**
